orderbook/src/orderbook.rs

```diff
@@ -123,9 +127,18 @@ impl Orderbook {
 
     pub async fn get_orders(&self, filter: &OrderFilter) -> Result<Vec<Order>> {
         let mut orders = self.database.orders(filter).try_collect::<Vec<_>>().await?;
-        set_order_balance(orders.as_mut_slice(), self.balance_fetcher.as_ref()).await;
+        let balances =
+            track_and_get_balances(self.balance_fetcher.as_ref(), orders.as_slice()).await;
+        // The meaning of the available balance field is different depending on whether we return
+        // orders for the solver or the frontend. For the frontend (else case) balances are always
+        // actual balances but for the solver there is custom logic to decide which orders get
+        // prioritized when a user's balance is too small to cover all of their orders.
+        // We can hopefully resolve this when we have a custom struct for orders in the
+        // get_solver_orders route and a custom endpoint to query user balances for the frontend.
         if filter.exclude_insufficient_balance {
-            remove_orders_without_sufficient_balance(&mut orders);
+            orders = solvable_orders(orders, &balances);
+        } else {
+            set_available_balances(orders.as_mut_slice(), &balances);
```

Comment:
Can we set the available balances regardless of whether or not we `exclude_insufficient_balance` orders? I frequently use the `/api/v1/orders/` endpoint and I think it would be nice to get non-0 values for available balance.

Or is there a reason we don't want to set that value?

Reply:
It should be set either way. If `exclude_insufficient_balance` is true then we go into the solvable_orders branch which sets it and if it is false we go into this branch which sets it.

Reply:
I'm probably blind, but I can't find where `solvable_orders` sets `available_balance`...

Reply:
You're right. I was convinced I did this in in solvable_orders but it's not in the diff. Fixed by moving `set_available_balances` out of the branch.